pipeline/matcher/scripts/getMatcherGraph.ts

```diff
@@ -0,0 +1,191 @@
+  const tableName = `catalogue-${input.pipelineDate}_works-graph`;
+
+  const output = await client.get({
+    TableName: tableName,
+    Key: { id: input.canonicalId }
+  });
+
+  if (typeof output.Item === 'undefined') {
+    throw new Error(`Could not find a matched work with ID ${input.canonicalId}`);
+  }
+
+  const subgraphId = output.Item!.subgraphId;
+
+  const query: QueryCommandInput = {
+    TableName: tableName,
+    IndexName: 'work-sets-index',
+    KeyConditionExpression: '#subgraphId = :subgraphId',
+    ExpressionAttributeNames: {'#subgraphId': 'subgraphId'},
+    ExpressionAttributeValues: {':subgraphId': subgraphId},
+  };
+
+  const worksInSubgraph: Array<Record<string, any>> = await queryAllMatchingItems(client, query);
+
+  return worksInSubgraph.map((item: Record<string, any>) => {
+    return {
+      canonicalId: item.id,
+      mergeCandidateIds: item.sourceWork.mergeCandidateIds,
+      suppressed: item.sourceWork.suppressed,
+      componentIds: item.componentIds,
+      sourceIdentifier: {
+        value: item.sourceWork.id.value,
+
+        // This is an artefact of a slightly weird format in DynamoDB,
+        // where the identifierType is recorded as e.g.
+        //
+        //    {"identifierType": {"METS": "METS"}, ...}
+        //
+        // We might simplify this at some point.
+        identifierType: Object.keys(item.sourceWork.id.identifierType)[0],
+      }
+    };
+  });
+}
+
+async function createGraph(works: SourceWork[]): Promise<RootCluster> {
+  const g = digraph('G',
+    
+    // This tells Graphviz to lay out the nodes left-to-right, because we tend
+    // to have monitors that are wider than tall, so it's easier to lay things
+    // out this way.
+    // See https://graphviz.org/docs/attrs/rankdir/
+    { rankdir: 'LR' },
+  );
+
+  // Create all the nodes
+  const nodes: Map<string, INode> = new Map();
+  await Promise.all(works.map(async (w: SourceWork) => {
+    let attributes = await getAttributes(w);
+
+    const newNode = g.createNode(w.canonicalId, attributes);
+
+    nodes.set(w.canonicalId, newNode);
+  }));
+
+  // Add all the edges
+  works.map((w: SourceWork) => {
+    w.mergeCandidateIds.forEach((target: string) => {
+      
+      // Make sure this node is in the graph.  If it's not in the graph, it means
+      // this Work has a merge candidate that points to a Work the matcher hasn't
+      // seen.  Show it on the graph, but make it clear we don't know what it is.
+      if (!nodes.has(target)) {
+        nodes.set(target, g.createNode(target, { style: 'dotted' }));
+      }
+
+      // If A->B but they're in different components, it means one or both of them
+      // is suppressed.  Include the edge, but make it clear it's not used.
+      let attributes = {};
+      if (!w.componentIds.includes(target)) {
+        attributes['style'] = 'dotted';
+      }
+
+      g.createEdge([nodes.get(w.canonicalId), nodes.get(target)], attributes);
+    });
+  });
+
+  return g;
```

Comment:
No need for this if the above is true I think

Reply:
I'm not sure I understand what you're suggesting? The variable is declared within this function, so we have to return it for callers to get to it.

Reply:
Misread the bracket matching!